### ISSUE TITLE
Custom Muted Sounds Not Working

### ISSUE THREAD

**Issue by Torchead** [Broooklyn/runelite-external-plugins#141]
Hello, I'm trying to mute the dwarf miner sounds at Motherlode Mine by entering the sound id's I pulled from the "visual sounds" plugin but I can't get it to work. Is there a different syntax I should be using?

![image](https://github.com/Broooklyn/runelite-external-plugins/assets/119005602/376973d2-4a5b-45a5-a1da-3000f0db3e7e)
![image](https://github.com/Broooklyn/runelite-external-plugins/assets/119005602/36ee7dd9-ac3d-419f-82a5-5311d009fce0)

**Comment by JZomDev** [comment]
Well it for sure wouldn't be the words of mine1,mine2 - but i am not familiar with visual sounds to know what it says is an ID. I imagine mine1,mine2 being in that list caused an issue. Did you get it to work?

**Comment by drownzz** [comment]
I have the same problem, seems like the custom mute option isn't working at all.
I have "Visual Sounds" enabled aswell and the presets from the "Annoyance mute" will be removed from the list if I toggle some options, for example the portal nexus. But none of the IDs I place in the custom ID option.

**Comment by JZomDev** [comment]
Can you screenshot what your config is set at in that list?

**Comment by drownzz** [comment]
For example this: 
In the list on the left you still see 2068 and 2188 (just picked those as an example)
![Image](https://github.com/user-attachments/assets/5355c431-d2f0-4035-bccc-d57d09189753)

**Comment by JZomDev** [comment]
You are using the ambient muting section instead of the normal sound muting section. Ambient sounds are like the sounds you hear when you get close to a fire or fountain or the bird chirps and wind you hear as you go into wooded areas. 

That'll fix your problem.

**Comment by drownzz** [comment]
I have added it in both of the sections, but still nothing

**Comment by drownzz** [comment]
Apparently it has something to do with sound id/name "-1". 
I have added that to the list and some sounds which I couldn't mute, are muted now.

I am unsure how to read the debug from the plugin, is the first number (before min/max) just a group including multiple sound ids?

**Comment by JZomDev** [comment]
So, ambient sounds are a master sound accompanied by other sounds, think wind as base sound and bird chirps as the extra sounds. 

There are some ambient sounds that use NO base line sound, so they use -1 and they have a following accompanied list of sounds to add to the silence. 

It wasn't clear to me that you were muting regular sounds like a pickaxe or a bird chirps. (I didn't actually see the screenshot properly, that's on me)

-1 will ambient mute much of the game and you maybe will be okay with that.

**Comment by JZomDev** [comment]
I wasn't aware of this plugin that adds the contents of the accompanied sound to the ambient sound primary, that's neat.

**Comment by drownzz** [comment]
I might try to remove the specific ambient sound id from the background sounds later this weekend and make a pull request. Should be possible

**Comment by JZomDev** [comment]
It is possible, I just didn't offer that because there wasn't a way to get the specific sound ids and have them named in a good way and they were effectively random to an end user.

**Comment by drownzz** [comment]
Can look at it together if you'd like?

**Comment by JZomDev** [comment]
I'm unable to for the next few days, if you want to try go ahead but I'll do it when I'm back.

Thanks for showing visual sounds does ambient sounds now, that's neat I had a crude way before jagex dumped the ids on us.

**Comment by drownzz** [comment]
Came to the conclusion that it is not possible to edit the backgroundSoundEffects on a AmbientSoundEffect. 
So either mute the AmbientSoundEffect totally if the ID is found in the backgroundSoundEffects or just leave it as it is.
Problem is that if you do mute it, you mute other sound effects aswell, which might not be ideal because you have no clue as a player which other sounds are getting muted.

If you come up with something else, let me know

**Comment by JZomDev** [comment]
I thought I had it, but it seems to not work with -1, and I'm not sure why. 


Maybe it can't be done. I'll try again though.

**Comment by drownzz** [comment]
Yeah, the backgroundSoundEffects is immutable and within a plugin you are not able to create a new object from AmbientSoundEffect with a setter for backgroundSoundEffects.
